C/common/include/config_category.h

```diff
@@ -126,7 +126,9 @@ class ConfigCategory {
 					MAXIMUM_ATTR,
 					LENGTH_ATTR,
 					VALIDITY_ATTR,
-					GROUP_ATTR};
+					GROUP_ATTR,
+					DISPLAY_NAME_ATTR,
+					DEPRECATED_ATTR};
```

Comment:
What about `rule` optional config item?

Reply:
When you say rule do you mean the validity rule? If so this is already covered as VALIDITY_ATTR.

Reply:
We have supported `rule` optional attribute as well. Validity optional attribute is different from it.
**rule**
```
The validation rule will take the form of an expression. If it evaluates to true then the proposed value of the configuration item will be allowed. If it evaluates to false then it will not be.
```

**validity**
```
An expression used to determine if the configuration item is valid. Used in the UI to gray out one value based on the value of others.
```

We have also added its documentation https://fledge-iot.readthedocs.io/en/develop/plugin_developers_guide/02_writing_plugins.html#configuration-lifecycle